src/rebar3_appup_generate.erl

```diff
@@ -206,15 +206,15 @@ generate_instruction(deleted, File) ->
 
 generate_instruction(changed, ModDeps, {File, _}) ->
     {ok, {Name, List}} = beam_lib:chunks(File, [attributes, exports]),
-    Behavior = get_behavior(List),
+    Supervisor = is_supervisor(List),
     CodeChange = is_code_change(List),
```

Comment:
In other behaviour, I think that there is nothing that it need to generate a special instruction.
As an exception, supecial process and supervisor behaviour will be able to co-exist.
(However, supecial process does not require a behaviour.)

Still do you want a change, such as the following?

```erlang
generate_instruction_advanced(Name, Behaviours, _, Deps) ->
    case lists:member(supervisor, Behaviours) of
        true  -> {update, Name, supervisor};
        false -> {load_module, Name, Deps}
end
```